tests/dummy/app/templates/docs/sharing-code.md

```diff
@@ -8,25 +8,13 @@ However, there are use cases for needing to share components and more across Eng
 
 ## Addon Re-exports
 
-In order to use things like Components and Helpers from addons in your Engine, they will need to be present in your Engine's namespace. This is accomplished by doing what is called a "re-export":
+Any addon included in your Engine's package.json `dependencies` key will automatically get included in your Engine.
```

Comment:
I think this section is talking about re-exports and this new change is talking about how dependencies are bundled in engines?

Reply:
That's the point, it does not make sense re-export on namespace `app`  components, services and etc. It breaks the isolation. 

Also the `require('ember-engines/lib/engine-addon')` is mandatory to ember-engines work. Then it doesn't make sense to put it as a config to import/export things.

Reply:
Anyway, or we keep this documentation or we should remove the section about `Engine Re-export`

Reply:
Right, for the sake of not using module unification or https://github.com/rwjblue/ember-holy-futuristic-template-namespacing-batman we should either point to a solution that doesn't require re-exports. But Re-exports work and are intended to solve a problem. I think we should keep this here for the time being.

Reply:
I don’t agree, It doesn’t make sense. We should avoid anti pattern in our documentation. Engine is an addon and because of this you can use the namespace `app`, but we should not recommend it 

What kind of problem re-export solve to ember engines?

Reply:
Having multiple engines and wanting to specify the non-namespaced path is a use case. If we don't want to suggest people to use re-exports, let's remove this section completely as the section no longer has information that makes sense.

Reply:
Done @gabrielcsapo 

I changed it to the proper section,

Reply:
@villander this only true if you have components in app and include those in your engine. If you include and addon with files in addon and not app, this is not accurate.

Reply:
@gabrielcsapo any addon can be used by an engine, if you have an addon, you can use it inside an engine. 

I can have an addon just with services and reuse it between engines and host apps.

Reply:
Can we explain how to use addons in your engine? Unless you use namespacing you have to reexport components under app under your addon to get them namespaced to your engine.

Reply:
What you only need at your Engine package at hand goes into **devDependencies** and what the consumers of your package need, too goes into **dependencies**